tests/factories/requests_.py

```diff
@@ -0,0 +1,50 @@
+import json
+from io import BytesIO
+
+import requests
+from factory import Factory, Faker
+
+
+class Response(Factory):
+    class Meta:
+        model = requests.Response
+
+    raw = None
+    json_data = None
+    encoding = "utf-8"
+    status_code = Faker(
+        "random_element", elements=[200, 201, 301, 304, 401, 404, 500, 501],
+    )
+    headers = None
+
+    @classmethod
+    def _adjust_kwargs(cls, **kwargs):
+        headers = kwargs["headers"] or {}
+        encoding = kwargs["encoding"]
+        raw = kwargs["raw"]
+        json_data = kwargs["json_data"]
```

Comment:
Pick your poison here. If we push these into the function signature, then we change the function signature, which prompts a warning. If we do this, it makes it hideously long... not sure which is worse.

Reply:
https://github.com/hypothesis/lms/pull/1946/ gets rid of this